salt/user-certs/init.sls

```diff
@@ -0,0 +1,44 @@
+include:
+  - crypto
+
+{% for cert in ['velum', 'salt-api'] %}
+/etc/pki/{{ cert }}.key:
+  x509.private_key_managed:
+    - bits: 4096
+    - require:
+      - sls:  crypto
+      - file: /etc/pki
+  file.managed:
+    - replace: false
+    - user: root
+    - group: root
+    - mode: 644
+
+/etc/pki/{{ cert }}.crt:
+  x509.certificate_managed:
+    - ca_server: {{ salt['mine.get']('roles:ca', 'x509.get_pem_entries', expr_form='grain').keys()[0] }}
+    - signing_policy: minion
+    - public_key: /etc/pki/{{ cert }}.key
+    - CN: {{ grains['fqdn'] }}
+    - C: {{ pillar['certificate_information']['subject_properties']['C']|yaml_dquote }}
+    - Email: {{ pillar['certificate_information']['subject_properties']['Email']|yaml_dquote }}
+    - GN: {{ pillar['certificate_information']['subject_properties']['GN']|yaml_dquote }}
+    - L: {{ pillar['certificate_information']['subject_properties']['L']|yaml_dquote }}
+    - O: {{ pillar['certificate_information']['subject_properties']['O']|yaml_dquote }}
+    - OU: {{ pillar['certificate_information']['subject_properties']['OU']|yaml_dquote }}
+    - SN: {{ pillar['certificate_information']['subject_properties']['SN']|yaml_dquote }}
+    - ST: {{ pillar['certificate_information']['subject_properties']['ST']|yaml_dquote }}
```

Comment:
About all the pillars used above (the `certificate_information`). I guess we will use the default value from [here](https://github.com/kubic-project/salt/blob/master/pillar/certificates.sls). Some of these values are empty, is this going to be a problem?